graylog2-web-interface/src/components/users/UserForm.jsx

```diff
@@ -111,6 +111,9 @@ const UserForm = React.createClass({
       if (this.isPermitted(this.state.currentUser.permissions, ['users:list'])) {
         this.props.history.replaceState(null, Routes.SYSTEM.AUTHENTICATION.USERS.LIST);
       }
+      if (this.props.user.username === this.state.currentUser.username) {
+        CurrentUserStore.reload();
+      }
```

Comment:
Yes, I think we need to. I would vote for refactoring the `UsersStore` into a Reflux store. :+1: